object_store_factory/src/local.rs

```diff
@@ -6,20 +6,33 @@ use std::sync::Arc;
 #[derive(Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct LocalConfig {
     pub data_dir: String,
+    #[serde(default = "default_false")]
+    pub no_hardlinks: bool,
```

Comment:
`no_hardlinks` implies the absence of hardlinks rather than an active setting. While `enable_feature_x` or  `disable_feature_y` indicates an action or a configuration setting.  

```suggestion
    pub disable_hardlinks: bool,
```

Reply:
The setting was named no_hardlinks to handle a system that doesn't actually support hard links. However, since it effectively disables the use of hard links, your feedback is valid. I will update it accordingly. Additionally, testing revealed that not all hard link functions have been bypassed yet. I will address those as well and share the updates once completed. Thank you for your review.